src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(hidden) {
+    super(hidden);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && Animal.alive.includes(prey)) {
+      if (!prey.hidden) {
+        prey.health -= 50;
+      }
+    }
+
+    if (prey.health === 0) {
+      Animal.alive.splice(Animal.alive.indexOf(prey), 1);
+    }
```

Comment:
from checklist
2. [CODE PERFOMANCE] - Try to avoid nesting loops (indexOf + splice), use filter instead